lib/ansible/modules/remote_management/manageiq/manageiq_dynamic_resource_instance.py

```diff
@@ -0,0 +1,381 @@
+      verify_ssl: False
+
+- name: Edit a dynamic resource definition in ManageIQ
+  manageiq_dynamic_resource_instance:
+    name: 'my_instance'
+    state: 'present'
+    property_attributes:
+      engine: 'mysql'
+      type: 'DB'
+      size: 30
+      region: 'nova'
+      username: 'masterdb'
+      services:
+        - service2
+      providers:
+        - Openstack
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+
+- name: Delete a dynamic resource definition in ManageIQ
+  manageiq_dynamic_resource_instance:
+    name: 'my_instance'
+    state: 'absent'
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+'''
+
+RETURN = '''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.manageiq import ManageIQ, manageiq_argument_spec
+
+
+class ManageIQDynamicResourceInstance(object):
+    """
+        Object to create dynamic resource instance operations in manageiq.
+    """
+
+    def __init__(self, manageiq):
+        self.manageiq = manageiq
+        self.module = self.manageiq.module
+        self.api_url = self.manageiq.api_url
+        self.client = self.manageiq.client
+
+    def dynamic_resource_definition(self, name):
+        """ Search for dynamic resource definition by name.
+
+        Returns:
+            the dynamic resource definition, or None if not found.
+        """
+        return self.manageiq.find_collection_resource_by('generic_object_definitions', name=name)
+
+    def dynamic_resource_instance(self, name):
+        """ Search for dynamic resource instance by name.
+
+        Returns:
+            the dynamic resource instance, or None if not found.
+        """
+        return self.manageiq.find_collection_resource_by('generic_objects', name=name)
+
+    def provider(self, name):
+        """ Search for provider by name.
+
+        Returns:
+            the provider, or None if not found.
+        """
+        return self.manageiq.find_collection_resource_by('providers', name=name)
+
+    def service(self, name):
+        """ Search for service by name.
+
+        Returns:
+            the service, or None if not found.
+        """
+        return self.manageiq.find_collection_resource_by('services', name=name)
+
+    def create_instance(
+        self, name, dynamic_resource_definition, property_attributes, services, providers
+    ):
+        """ Creates the dynamic resource instance in manageiq.
+
+        Returns:
+            a dictionary with changes and msg.
+        """
+        url = '%s/generic_objects' % (self.api_url)
+
+        generic_object_definition_href = '%s/generic_object_definitions/%s' % (
+            self.api_url,
+            dynamic_resource_definition['id']
+        )
+
+        resource = {
+            'name': name,
+            # API still uses old name
```

Comment:
did we get any answer which name is right?  what does UI use?

Reply:
All the docs have been renamed
https://bugzilla.redhat.com/show_bug.cgi?id=1506407
And the PR has been merged
https://github.com/ManageIQ/manageiq_docs/pull/553
I think it's safe to say that it's the new name